testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -1063,6 +1063,40 @@ fn wait_for_block_global_rejection(
     })
 }
 
+/// Waits for >30% of num_signers block rejection to be observed in the test_observer stackerdb chunks for a block
+/// with the provided signer signature hash and the specified reject_code
+fn wait_for_block_global_rejection_with_reject_code(
+    timeout_secs: u64,
+    block_signer_signature_hash: Sha512Trunc256Sum,
+    num_signers: usize,
+    reject_code: RejectCode,
+) -> Result<(), String> {
+    let mut found_rejections = HashSet::new();
+    wait_for(timeout_secs, || {
+        let chunks = test_observer::get_stackerdb_chunks();
+        for chunk in chunks.into_iter().flat_map(|chunk| chunk.modified_slots) {
+            let Ok(message) = SignerMessage::consensus_deserialize(&mut chunk.data.as_slice())
+            else {
+                continue;
+            };
+            if let SignerMessage::BlockResponse(BlockResponse::Rejected(BlockRejection {
+                signer_signature_hash,
+                signature,
+                reason_code,
+                ..
```

Comment:
You can get the `RejectReason` from the `response_data` here if that would be better.

Reply:
Definitely better! thanks a lot